web/src/user_group_edit.ts

```diff
@@ -1540,28 +1542,59 @@ export function maybe_reset_right_panel(groups_list_data: UserGroup[]): void {
 export function update_empty_left_panel_message(): void {
     // Check if we have any groups in panel to decide whether to
     // display a notice.
-    let has_groups;
     const is_your_groups_tab_active =
         get_active_data().$tabs.first().attr("data-tab-key") === "your-groups";
-    if (is_your_groups_tab_active) {
-        has_groups = user_groups.get_user_groups_of_user(people.my_current_user_id()).length;
-    } else {
-        has_groups = user_groups.get_realm_user_groups().length;
+
+    let current_group_filter =
+        z.string().optional().parse(filters_dropdown_widget.value()) ??
+        FILTERS.ACTIVE_AND_DEACTIVATED_GROUPS;
+
+    // When the dropdown menu is hidden.
+    if ($("#user-group-edit-filter-options").is(":hidden")) {
+        current_group_filter = FILTERS.ACTIVE_AND_DEACTIVATED_GROUPS;
     }
-    if (has_groups) {
+
+    if ($(".user-groups-list").find(".group-row:visible").length > 0) {
         $(".no-groups-to-show").hide();
         return;
     }
+
+    const empty_user_group_list_message = get_empty_user_group_list_message(
+        current_group_filter,
+        is_your_groups_tab_active,
+    );
+
+    const args = {
+        empty_user_group_list_message,
+        can_create_user_groups:
+            settings_data.user_can_create_user_groups() && realm.zulip_plan_is_not_limited,
```

Comment:
Probably a good follow-up refactor will be to make `user_can_create_user_groups` check `realm.zulip_plan_is_not_limited` itself, since I expect all other callers will want that behavior, but that should be its own PR since it does need careful checking of the other code paths.